src/esp/sensor/SensorFactory.cpp

```diff
@@ -0,0 +1,18 @@
+#include "esp/sensor/SensorFactory.h"
+
+#include "esp/scene/SceneNode.h"
+#include "esp/sensor/CameraSensor.h"
+#include "esp/sensor/Sensor.h"
+
+namespace esp {
+sensor::SensorSuite SensorFactory::createSensors(
+    scene::SceneNode& node,
+    const sensor::SensorSetup& sensorSetup) {
+  sensor::SensorSuite sensorSuite = sensor::SensorSuite();
+  for (sensor::SensorSpec::ptr spec : sensorSetup) {
```

Comment:
Fix this loop using the following code : 
`for (const sensor::SensorSpec::ptr& spec : sensorSetup)`